resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1172,6 +1181,31 @@ private[spark] class Client(
     }.mkString("")
   }
 
+  private def getDriverLogsLink(appId: String): Option[(String, String)] = {
+    val baseRmUrl = WebAppUtils.getRMWebAppURLWithScheme(hadoopConf)
+    val response = ClientBuilder.newClient()
+      .target(baseRmUrl)
+      .path("ws").path("v1").path("cluster").path("apps").path(appId).path("appattempts")
+      .request(MediaType.APPLICATION_JSON)
+      .get()
+    response.getStatusInfo.getFamily match {
+      case Family.SUCCESSFUL =>
+        val objectMapper = new ObjectMapper()
+        // If JSON response is malformed somewhere along the way, MissingNode will be returned,
+        // which allows for safe continuation of chaining. The `elements()` call will be empty,
+        // and None will get returned.
+        objectMapper.readTree(response.readEntity(classOf[String]))
+            .path("appAttempts").path("appAttempt")
+            .elements().asScala.toList.takeRight(1).headOption
+            .map(_.path("logsLink").asText())
+            .map(baseUrl => (s"$baseUrl/stdout?start=-4096", s"$baseUrl/stderr?start=-4096"))
+      case _ =>
+        logInfo(s"Unable to fetch app attempts info from $baseRmUrl, got "
```

Comment:
Should this be at Info level?

Reply:
Do you think it should be WARN? I waffled on this given that it is a very non-critical code path.

Reply:
Yes because when I look at any errors during fetch of application report, they are logged at error level. Ideally if that succeeds this should as well. If it doesn't, I think it should be logged at warn level.

Reply:
Makes sense, I will upgrade to WARN.